src/utilities/codemirror/vml-linter.ts

```diff
@@ -0,0 +1,288 @@
+    } else if (!argNode && !!argDef) {
+      const { from, to } = functionNameNode;
+      diagnostics.push({
+        from,
+        message: `${functionName} is missing argument ${argDef.name}`,
+        severity: 'error',
+        to,
+      });
+    }
+  }
+  const extraArgs = parametersNode.slice(commandDef.arguments.length);
+  diagnostics.push(
+    ...extraArgs.map((extraArg: SyntaxNode): Diagnostic => {
+      const { from, to } = extraArg;
+      return {
+        from,
+        message: `${functionName} has an extra argument ${docText.slice(from, to)}`,
+        severity: 'error',
+        to,
+      };
+    }),
+  );
+  return diagnostics;
+}
+
+function validateArgument(
+  commandDictionary: CommandDictionary,
+  argDef: FswCommandArgument,
+  argNode: SyntaxNode,
+  docText: string,
+): Diagnostic[] {
+  const diagnostics: Diagnostic[] = [];
+
+  // could also be a variable
+  const constantNode = argNode.getChild('Simple_expr')?.getChild('Constant')?.firstChild;
+
+  if (constantNode) {
+    const { from, to } = constantNode;
+    switch (argDef.arg_type) {
+      case 'integer':
+        {
+          if (![TOKEN_INT_CONST, TOKEN_UINT_CONST, TOKEN_HEX_CONST].includes(constantNode.name)) {
+            return [
+              {
+                from,
+                message: `Expected integer value`,
+                severity: 'error',
+                to,
+              },
+            ];
+          } else if (argDef.range) {
+            // TODO: CDL dictionary provides a conversion, HEX arguments should prefer hexadecimal
+            const base = constantNode.name === TOKEN_HEX_CONST ? 16 : 10;
+            const argValue = parseInt(docText.slice(argNode.from, argNode.to), base);
+            if (argValue < argDef.range.min || argValue > argDef.range.max) {
+              return [
+                {
+                  from,
+                  message: `Value should be between ${argDef.range.min.toString(base)} and ${argDef.range.max.toString(base)}`,
+                  severity: 'error',
+                  to,
+                },
+              ];
+            }
+          }
+        }
+        break;
+      case 'float':
+        if (![TOKEN_INT_CONST, TOKEN_DOUBLE_CONST].includes(constantNode.name)) {
+          return [
+            {
+              from,
+              message: `Expected float or integer value`,
+              severity: 'error',
+              to,
+            },
+          ];
+        } else if (argDef.range) {
+          const argValue = parseFloat(docText.slice(argNode.from, argNode.to));
+          if (argValue < argDef.range.min || argValue > argDef.range.max) {
+            return [
+              {
+                from,
+                message: `Value should be between ${argDef.range.min.toString()} and ${argDef.range.max.toString()}`,
+                severity: 'error',
+                to,
+              },
+            ];
+          }
+        }
+        break;
+      case 'var_string':
```

Comment:
add` case 'fixed_string':`

Reply:
addressed in https://github.com/NASA-AMMOS/aerie-ui/pull/1472/commits/63dee070ed71cd3a6c463d392b114c98ac5577aa